extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,21 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    const selector = config['selector'] || (spec && spec['selector']);
+    const analyticsVariables = config[vars] || {};
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
+      const el = this.win_.document.getElementById(selector
```

Comment:
I had the same feeling that we are doing the DOM reference in 2 place, one in [instrumentation.js](https://github.com/rajkumarsrk/amphtml/blob/master/extensions/amp-analytics/0.1/instrumentation.js#L223) and other on [visibility-impl.js](https://github.com/rajkumarsrk/amphtml/blob/master/extensions/amp-analytics/0.1/visibility-impl.js#L245). But when you check the implementation done for the 'click' event we have the element level overrides extracted on the `[createSelectiveListener_](https://github.com/rajkumarsrk/amphtml/blob/master/extensions/amp-analytics/0.1/instrumentation.js#L315)`. And we have access to `extractAnalyticsVariables_` inside `instrumentation.js` and not on `visibility-impl.js`. Let me know your thoughts on this